### PR TITLE
Implement presences for `Guild.query_members`

### PR DESCRIPTION
**Pull request by NCPlayz**
### Summary

This implements discordapp/discord-api-docs#1116.

### Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by NotSoSuper** [comment]
update

File: discord/guild.py [hunk 1]
```diff
@@ -2156,9 +2159,8 @@ async def query_members(self, query=None, *, limit=5, user_ids=None, cache=True)
 
         if user_ids is not None and query is not None:
             raise ValueError('Cannot pass both query and user_ids')
-
-        limit = min(100, limit or 5)
```
Comment (Rapptz) [inline discord/guild.py:2160]: Why was this removed?

File: discord/guild.py [hunk 2]
```diff
@@ -2125,6 +2125,9 @@ async def query_members(self, query=None, *, limit=5, user_ids=None, cache=True)
         limit: :class:`int`
             The maximum number of members to send back. This must be
             a number between 5 and 100.
+        presences: :class:`bool`
+            Whether to request for presences to be provided. This defaults
+            to ``False``.
```
Comment (Rapptz) [inline discord/guild.py:2130]: Needs versionadded

File: discord/guild.py [hunk 3]
```diff
@@ -2108,7 +2108,7 @@ async def chunk(self, *, cache=True):
 
         return await self._state.chunk_guild(self, cache=cache)
 
-    async def query_members(self, query=None, *, limit=5, user_ids=None, cache=True):
+    async def query_members(self, query=None, *, limit=5, user_ids=None, presences=False, cache=True):
```
Comment (Rapptz) [inline discord/guild.py:2111]: This should probably check for the presence intent.

File: discord/state.py [hunk 4]
```diff
@@ -967,8 +967,19 @@ def parse_guild_role_update(self, data):
...
         members = [Member(guild=guild, data=member, state=self) for member in data.get('members', [])]
         log.debug('Processed a chunk for %s members in guild ID %s.', len(members), guild_id)
+
+        if presences:
+            member_dict = {str(member.id): member for member in members}
+            for presence in presences:
+                user = presence['user']
+                member_id = user['id']
+                member = member_dict.get(member_id)
+                member._presence_update(presence, user)
```
Comment (Rapptz) [inline discord/state.py:981]: Since the `presences` key will mostly be missing, use a `try except KeyError` instead. 

Might be worth swapping the lookup table (`{user_id: presence}` instead of `{user_id: member}`) though this point is minor.

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/guild.py [hunk 5]
```diff
@@ -2143,13 +2146,18 @@ async def query_members(self, query=None, *, limit=5, user_ids=None, presences=F
...
+        ClientException
+            The presences intent is not enabled.
 
         Returns
         --------
         List[:class:`Member`]
             The list of members that have matched the query.
         """
 
+        if not self._state._intents.presences:
```
Comment (Rapptz) [inline discord/guild.py:2158]:
```suggestion
        if presences and not self._state._intents.presences:
```

**Review by Rapptz — CHANGES_REQUESTED** [review]

**Review by Rapptz — APPROVED** [review]
